### PR TITLE
Use dial with timeout when acquiring new connection

### PR DESCRIPTION
**Pull request by mariokostelac**

### CODE REVIEW

**Comment by bscanlan** [comment]
:ok_man:

**Comment by mariokostelac** [comment]
Confirmed on one box that it works correctly.

File: proxy.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func (p *Proxy) AuthConn(conn net.Conn) error {
 func (p *Proxy) newServerConn() (io.Closer, error) {
 	retrySleep := 50 * time.Millisecond
 	for retryCount := 7; retryCount > 0; retryCount-- {
-		c, err := net.Dial("tcp", p.MongoAddr)
+		c, err := net.DialTimeout("tcp", p.MongoAddr, time.Second)
```
Comment (dongintercom) [inline proxy.go:131]: does `time.Second` mean 1 second here?

Comment (mariokostelac) [reply proxy.go:131]: yes

**Pull request merged**